sheepdog/transactions/upload/transaction.py

```diff
@@ -172,7 +178,43 @@ def flush(self):
         """
         for entity in self.valid_entities:
             entity.flush_to_session()
-        self.session.flush()
+        try:
+            self.session.flush()
+        except IntegrityError as e:
+            # don't handle non-unique constraint errors
+            if "duplicate key value violates unique constraint" not in e.message:
+                raise
+            values = VALUES_REGEXP.findall(e.message)
+            if not values:
+                raise
+            values = [v.strip() for v in values[0].split(",")]
+            keys = KEYS_REGEXP.findall(e.message)
+            if len(keys) == len(values):
+                values = dict(zip(keys, values))
+                entities = []
+                label = None
+                for en in self.valid_entities:
+                    for k, v in values.items():
+                        if getattr(en.node, k, None) != v:
+                            break
+                    else:
+                        if label and label != en.node.label:
+                            break
+                        entities.append(en)
+                        label = en.node.label
+                else:  # pylint: disable=useless-else-on-loop
+                    # https://github.com/PyCQA/pylint/pull/2760
+                    for entity in entities:
+                        entity.record_error(
+                            "{} with {} already exists in the GDC".format(
```

Comment:
while we are here, can we change this to just be `{} with {} already exists` so its more agnostic?